_tools/utils.ts

```diff
@@ -34,3 +36,37 @@ export function resolveWorkspaceSpecifiers(
     return new URL(specifier).href;
   }
 }
+
+/**
+ * Checks whether a file is named `test.ts` and has no exports.
+ * @param filePath
+ * @returns true if file is named `test.ts` and has no exports, false otherwise
+ */
+export function isTestFile(filePath: string): boolean {
+  if (!filePath.endsWith("test.ts")) return false;
+  const source = Deno.readTextFileSync(filePath);
+  const sourceFile = ts.createSourceFile(
+    filePath,
+    source,
+    ts.ScriptTarget.Latest,
+  );
+
+  let result = true;
+
+  function visitNode(node: ts.Node) {
+    if (!result) return;
+    if (
+      ts.isExportSpecifier(node) ||
+      ts.isExportAssignment(node) ||
+      ts.isExportDeclaration(node) ||
+      node.kind === ts.SyntaxKind.ExportKeyword
+    ) {
+      result = false;
+    } else {
+      ts.forEachChild(node, visitNode);
+    }
+  }
+
+  visitNode(sourceFile);
+  return result;
+}
```

Comment:
This seems extraneous. What about this?
```ts
export function isTestFile(filePath: string): boolean {
  return filePath.endsWith("test.ts");
}
```

Reply:
The whole point of this function is to distinguish between `test.ts` which is used for testing (calling `Deno.test()`) and `test.ts` which is part of an api (like in `@std/front-matter/test`.  This cannot be done by filename alone. I figured the best way was to check whether the file has exports or not, if so it must be part of an api.

Reply:
Maybe `Deno.readTextFileSync(filePath).includes("\nDeno.test")` would suffice? (leading `\n` ensures it's at the beginning of the line)

Reply:
This could lead to unintentional behaviour if `Deno.test` is mentioned in a comment or used in an API. So special treatment would be required again for at least `@std/testing`.

Reply:
This could lead to unintentional behaviour if `Deno.test` was wrapped in a function and therefore not at the beginning of the line, for example (`jsonc/testdata/JSONTestSuite/test.ts`)
Also leads to unintentional behaviour when mentioned in a comment with unlucky line breaks:
```
/* bla bla bla...
Deno.test bla bla bla...
bla bla bla... */
```
So while this might work sometimes, it is not very robust.